pkg/ottl/expression.go

```diff
@@ -481,7 +493,8 @@ func (p *Parser[K]) newGetter(val value) (Getter[K], error) {
 	if val.Enum != nil {
 		enum, err := p.enumParser(val.Enum)
 		if err != nil {
-			return nil, err
+			// This is a hack, which assumes a literal function name when the enum parsing fails
+			return &literal[K]{value: string(*val.Enum)}, nil
```

Comment:
Does this actually work?  When invoked by `functions.go` this will return a `literal` which would then be used as the value for `StandardFunctionGetter`.  I expected somewhere we'd need to invoke `newFunctionCall`.

Reply:
What  I'm trying to do here is to return the string equivalent of the converter name which is passed as the fourth parameter. The idea is one could then use the function name (SHA256) and the argument (replacement) to make a function call as shown in `replace_pattern` below:

```
func replacePattern[K any](target ottl.GetSetter[K], regexPattern string, **replacement ottl.StringGetter[K]**, **hashFunction ottl.FunctionGetter[K]**) (ottl.ExprFunc[K], error) {
	return func(ctx context.Context, tCtx K) (interface{}, error) {
                 // this returns an ottl.ExprFunc
		function, err := hashFunction.Get(ctx, tCtx)
                // This returns the function name (string) which in this case is SHA256
		hashfunction, err := function(ctx, tCtx)
               // The goal is to then call the function with a StringGetter parameter (passed as an arg above)
               replacementHashValue := Call(hashfunction.(string), replacement) // Call a function by its string name using reflection
}
```
Perhaps, instead of resolving the function name as a literal, I should be passing it as a converter to a `newFunctionCall` instead, but I'm not sure how to fetch its argument which is the third parameter in `replace_pattern`.

Reply:
Ok I think I understand what you're saying. I forgot that Functions need their arguments during parsing and when we call the function we are in hot path execution. 

I think there is probably a way to handle this within FunctionGetter implementation. Instead of FunctionGetter returning a function for the caller to invoke later, maybe FunctionGetter should be more like a wrapper. When invoked, the FunctionGetter (maybe better name would be FunctionInvoker) would handle calling the function referenced in the OTTL statement and return its returned value. 

I expect FunctionInvoker would save the reference function name during parsing and then during hot path execution could be given a slice of arguments and then call newFunctionCall and then immediately invoke the result. This is inefficient, but will allow us to encapsulate the actual invoking. I don't think we want users of FunctionGetter to have to worry about reflection.

Reply:
ok makes sense. I'll implement a wrapper that invokes the passed in function with its arguments. I think I'll need to initialize an `editor` with the parsed function name and its arguments and then invoke `newFunctionCall` with the `editor`.

Reply:
@TylerHelmuth @evan-bradley to summarize, I'm re-using the parser, to initialize the editors/converters. This allows us to construct a statement (string) like the following - funcName(arg1, arg2,...) and leverage the existing grammar for lexical analysis followed by parsing. Please let me know if you think this makes sense or is a feasible approach. Open to trying something else too.

Reply:
Now that I see the other approach [here](https://github.com/evan-bradley/opentelemetry-collector-contrib/commit/b6edc4b483f5d59aa79a0e4c745ff48ae3004bb7#diff-073258313a2dc02137ad7eaeaf7323b6bf9639115905c2532c65bb635046229bR132), I prefer it since we're prepping the function with its arguments before and only calling the function in the hot path.